frontend/src/App.js

```diff
@@ -1,163 +1,204 @@
-    }
-
-    render() {
-        return (
-            <div className="App">
-                <header className="App-header">
-                    <h1>Instagram++</h1>
-                </header>
-                <Router>
-                    <div>
-                        {this.state.currentUser ?
-                            <div>
-                                <p>Logged in as: <Link to={`/account/${this.state.currentUser.username}`}>{this.state.currentUser.username}</Link></p>
-                                <Link to="/logout">Logout</Link><br/>
-                                <Link to="/post">Post Picture</Link><br/>
-                            </div>
-                            :
-                            <div>
-                                <Link to="/login">Login</Link><br/>
-                                <Link to="/register">Register</Link><br/>
-                            </div>
-                        }
-
-                        <Link to={`/post/${this.state.post_id}`}>Post #</Link>
-                        <input value={this.state.post_id} onChange={this.handleChangePostId}/><br/>
-
-                        <Link to={`/account/${this.state.username}`}>Search user</Link>
-                        <input value={this.state.username} onChange={this.handleChangeUser}/>
-                    </div>
-                    <hr/>
-                    <Switch>
-                        <Route exact path="/register" render={(props) => this.state.currentUser ? <Redirect to='/'/> :
-                            <Register {...props} onSuccess={this.login.bind(this)}/>
-                        }/>
-                        <Route exact path="/login" render={(props) => this.state.currentUser ? <Redirect to='/'/> :
-                            <Login {...props} onSuccess={this.login.bind(this)}/>
-                        }/>
-                        <Route exact path="/logout" render={() => {
-                            this.logout();
-                            return <Redirect to='/'/>;
-                        }}/>
-                        <Route exact path="/post" render={(props) => {
-                            return this.state.currentUser ? <PostPicture {...props}/> :  <Redirect to='/'/>;
-                        }}/>
-                        <Route path="/post/:id" render={({match}) => (<Post id={match.params.id}/>)}/>
-                        <Route path="/account/:username" render={({match}) => (<User username={match.params.username}/>)}/>
-                    </Switch>
-                </Router>
-            </div>
-        );
-    }
+export const App = () => {
+	const [username, setUsername] = useState();
+	const [currentUser, setCurrentUser] = useState();
+
+	useEffect(() => {
+		if (localStorageService.getAccessToken()) {
+			localStorageService.setBearerToken();
+		}
+		setLoggedInState();
+	}, [])
+
+	const setLoggedInState = () => {
+		axios.get(global.config.BACKEND_URL + "/account")
```

Comment:
Move setLoggedInState to inside the useEffect to prevent a warning.

Reply:
Unless it has to be outside to use it outside of useEffect too? I wonder if there is a workaround to avoid the warning and be able to use it elsewhere... Might be worth looking into.

Reply:
Can I ask what warning you're referring to? Also, would that mean also putting the `login/logout` in the useEffect too? Because if I move the entire `setLoggedInState()` to useEffect, then `login/logout`, which both have calls to `setLoggedInState()` throw me errors saying it isn't defined...

Reply:
When we call a function in useEffect that isn't defined inside of it we get a missing dependency warning. I don't think its a big deal though.